### PR TITLE
Gemfile refactorings

### PR DESCRIPTION
**Pull request by stefan-kolb**
In preparation for adding Oculus crawler logic to versioneye 

- Simplify Gemfile
- Use Version.rb for version definition
- Remove Gemfile.lock

These are all just suggestions, I understand if you don't want any of those changes.
Maybe we can also schedule a call so I can clarify why I suggest these changes and can get some more insight into the Versioneye system to integrate Oculus.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,9 +3,8 @@ source "http://rubygems.org"
 gem 'bundler'            , '~> 1.12.5'
 gem 'naturalsorter'      , '~> 3.0.14'
 gem 'dalli'              , '~> 2.7.0'
-platforms :ruby do
-  gem 'kgio'             , '~> 2.10.0'   # 20% performance boost for dalli
-end
+# 20% performance boost for dalli
+gem 'kgio'               , '~> 2.10.0', platforms: :ruby
```
Comment (reiz) [inline Gemfile]: That makes sense. I like the shorter version.

**Comment by reiz** [comment]
Hey @stefan-kolb. Many Thanks for your pull request. Do you know the [jeweler](https://www.versioneye.com/ruby/jeweler/2.1.1) project? We are using it to manage this Gem project. If I create a new patch release I use this script for example: https://github.com/versioneye/versioneye-core/blob/master/scripts/patch.sh. So all the entries in the gemspec are auto generated. 

I thought about skipping the Gemfile.lock and at first it was OK for me. But now I would like to keep it because I monitor the Gemfile.lock with VersionEye and if there are some security vulnerabilities in the transitive dependencies I would like to know about it.

What is the value of moving the depdencies from the Gemfile to the gemspec? At least for right now I would like to keep them in the Gemfile because I can monitor the Gemfile with VersionEye. Right now we don't have support to monitor gemspec files. That will come in future.

**Comment by stefan-kolb** [comment]
Hi @reiz,
I know the jeweler project. I get the idea why you use it, but in my opinion it adds unecessary overhead and it is easy to just use the build in Gem functionality of bundler or just `gem build` . In your case the gemspec is overly big including all files and need to be regenerated all the time. In my case you don't need to update anything except the version number for a new release (which is typically done inside this `version.rb` which is not used at the moment). This is maybe the only drawback as you can do this automatically at the moment. However, jeweler is not the real issue here, we could keep this the way it is.

A bigger problem is the Gemfile.lock and the overly strict dependencies. For a rationale on why I moved the dependencies to the gemspec and removed the Lockfile, see http://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/.
Problems here are the strict dependencies that may force conflicts in other application setups and the Lockfile which makes it for example impossible to run this Gem on Windows.

Actually, the project is no real Gem at the moment. The question is if you want to make it a Gem or not. 
Again, we can set up a call so I can describe the things I had in mind more easily. Just drop me a mail stefan.kolb(at)uni-bamberg.de.

**Comment by stefan-kolb** [comment]
For parsing gemspecs you could just use https://github.com/gemnasium/gemnasium-parser from your competitor :wink: ?!

**Comment by reiz** [comment]
Hi @stefan-kolb, I just finished reading the article. Now it makes sense to me :)

**Comment by reiz** [comment]
@stefan-kolb Integrating the gemnasium parser would be another thread. You wanna work on it?

**Comment by stefan-kolb** [comment]
@reiz Maybe at a later time. First priority for me would be to integrate the crawling logic for runtime languages as I want to use it inside https://paasfinder.org and deprecate my old crawler.

**Comment by reiz** [comment]
Hey @stefan-kolb. All right. That's OK for me. 

Since I merged the pull request I can't start the ruby console anymore. 

```
rake console --trace
W, [2016-07-11T17:35:17.213465 #80648]  WARN -- : Creating scope :parents. Overwriting existing method Project.parents.
initialize Settings
load config/settings.json
rake aborted!
Don't know how to build task 'console' (see --tasks)
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/gems/rake-11.2.2/lib/rake/task_manager.rb:71:in `[]'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/gems/rake-11.2.2/lib/rake/application.rb:151:in `invoke_task'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/gems/rake-11.2.2/lib/rake/application.rb:108:in `block (2 levels) in top_level'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/gems/rake-11.2.2/lib/rake/application.rb:108:in `each'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/gems/rake-11.2.2/lib/rake/application.rb:108:in `block in top_level'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/gems/rake-11.2.2/lib/rake/application.rb:117:in `run_with_threads'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/gems/rake-11.2.2/lib/rake/application.rb:102:in `top_level'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/gems/rake-11.2.2/lib/rake/application.rb:80:in `block in run'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/gems/rake-11.2.2/lib/rake/application.rb:178:in `standard_exception_handling'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/gems/rake-11.2.2/lib/rake/application.rb:77:in `run'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/gems/rake-11.2.2/exe/rake:27:in `<top (required)>'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/bin/rake:23:in `load'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/bin/rake:23:in `<main>'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/bin/ruby_executable_hooks:15:in `eval'
/Users/reiz/.rvm/gems/ruby-2.3.0@veye-core/bin/ruby_executable_hooks:15:in `<main>'
```

Any thoughts on this? 

I have another branch where I can do hot fixes right now. In the long run I want to make this project to a real Gem project that's why I will go through this changes.

**Comment by stefan-kolb** [comment]
Probably the console task came from jeweler, so it is gone right now. Maybe we have to add one manually if you need it: https://simonecarletti.com/blog/2009/09/rake-console/

**Comment by reiz** [comment]
@stefan-kolb The tests on CircleCI are all green after the merge. But somehow the `console` task disappeared from `rake`. Maybe it was part of jeweler?

**Comment by reiz** [comment]
@stefan-kolb Yep, that worked. Thanks for your help.

**Pull request merged**